src/instrument.ts

```diff
@@ -0,0 +1,262 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  SEMATTRS_DB_NAME,
+  SEMATTRS_DB_STATEMENT,
+  SEMATTRS_DB_SYSTEM,
+  SEMATTRS_DB_SQL_TABLE,
+} from '@opentelemetry/semantic-conventions';
+
+const {TracerProvider} = require('@opentelemetry/sdk-trace-node');
+
+import {
+  ContextManager,
+  Span,
+  SpanStatusCode,
+  context,
+  trace,
+  INVALID_SPAN_CONTEXT,
+  SpanAttributes,
+  TimeInput,
+  Link,
+  Exception,
+  SpanContext,
+  SpanStatus,
+  SpanKind,
+} from '@opentelemetry/api';
+
+let optedInPII: boolean =
+  process.env.SPANNER_ENABLE_EXTENDED_TRACING === 'true';
+
+interface SQLStatement {
+  sql: string;
+}
+
+interface observabilityOptions {
+  tracerProvider: typeof TracerProvider;
+  enableExtendedTracing: boolean;
+}
+
+export type {observabilityOptions as ObservabilityOptions};
+
+export function applyObservabilityOptions(
+  opts: observabilityOptions | undefined
+) {
+  if (!opts) {
+    return;
+  }
+
+  if (opts.tracerProvider) {
+    setTracerProvider(opts.tracerProvider);
+  }
+
+  if (opts.enableExtendedTracing) {
+    optInForSQLStatementOnSpans();
+  }
+}
+
+let defaultTracerProvider: typeof TracerProvider = undefined;
+
+// setTracerProvider allows the caller to hook up an OpenTelemetry
+// TracerProvider that spans generated from this library shall be attached to,
+// instead of using the global configuration. Later on if `getTracer` is invoked and
+// the default tracerProvider is unset, it'll use the global tracer
+// otherwise it'll use the set TracerProvider.
+export function setTracerProvider(freshTracerProvider: typeof TracerProvider) {
+  defaultTracerProvider = freshTracerProvider;
+}
+
+const TRACER_NAME = 'nodejs-spanner';
+
+// getTracer fetches the tracer each time that it is invoked to solve
+// the problem of observability being initialized after Spanner objects
+// have been already created.
+export function getTracer(config?: traceConfig) {
+  if (config && config.opts && config.opts.tracerProvider) {
+    return config.opts.tracerProvider.getTracer(TRACER_NAME);
+  }
+  if (defaultTracerProvider) {
+    return defaultTracerProvider.getTracer(TRACER_NAME);
+  }
+  // Otherwise use the global tracer.
+  return trace.getTracer(TRACER_NAME);
+}
+
+// optInForSQLStatementOnSpans is a configurable knob that if
+// invoked allows spans to be annotated with the SQL statement
+// of the producing function; if optOutofSQLStatementOnSpans
+// is invoked the SQL statement annotation shall be dropped.
+export function optInForSQLStatementOnSpans() {
+  optedInPII = true;
+}
+
+export function optOutOfSQLStatementOnSpans() {
+  optedInPII = false;
+}
+
+interface traceConfig {
+  sql?: string | SQLStatement;
+  tableName?: string;
+  dbName?: string;
+  opts?: observabilityOptions;
+  enableExtendedTracing?: boolean;
```

Comment:
Why do we need it again here ? We are already passing `observabilityOptions`

Reply:
Fixed and thank you @surbhigarg92!